source/vistas/core/graphics/overlay.py

```diff
@@ -0,0 +1,324 @@
+
+
+class Button(wx.EvtHandler):
+    """ A button to be drawn on the overlay """
+
+    DEFAULT = 'default'
+    HOVER = 'hover'
+
+    def __init__(self, position, size, default_image, hover_image=None):
+        super().__init__()
+
+        self.overlay = None
+        self.uv = None
+        self.state = self.DEFAULT
+
+        self._position = position
+        self._size = size
+        self._default_image = None
+        self._hover_image = None
+
+        self.default_image = default_image
+        self.hover_image = hover_image or default_image
+
+        self.Bind(wx.EVT_ENTER_WINDOW, self.on_mouse_enter)
+        self.Bind(wx.EVT_LEAVE_WINDOW, self.on_mouse_leave)
+        self.Bind(wx.EVT_LEFT_DOWN, self.on_mouse_click)
+
+    @property
+    def position(self):
+        return self._position
+
+    @position.setter
+    def position(self, position):
+        if self._position == position:
+            return
+
+        self._position = position
+        if self.overlay:
+            self.overlay.needs_mesh_update = True
+
+    @property
+    def size(self):
+        return self._size
+
+    @size.setter
+    def size(self, size):
+        if self._size == size:
+            return
+
+        self._size = size
+        if self.overlay:
+            self.overlay.needs_mesh_update = True
+
+    @property
+    def default_image(self):
+        return self._default_image
+
+    @default_image.setter
+    def default_image(self, image):
+        if image is self._default_image:
+            return
+
+        self._default_image = image
+        if self.overlay:
+            self.overlay.reset()
+
+    @property
+    def hover_image(self):
+        return self._hover_image
+
+    @hover_image.setter
+    def hover_image(self, image):
+        if image is self._hover_image:
+            return
+
+        self._hover_image = image
+        if self.overlay:
+            self.overlay.reset()
+
+    def on_mouse_enter(self, event):
+        self.state = self.HOVER
+
+        if self.overlay:
+            self.overlay.needs_mesh_update = True
+
+        self.overlay.canvas.Refresh()
+
+    def on_mouse_leave(self, event):
+        self.state = self.DEFAULT
+
+        if self.overlay:
+            self.overlay.needs_mesh_update = True
+
+        self.overlay.canvas.Refresh()
+
+    def on_mouse_click(self, event):
+        wx.PostEvent(self, wx.CommandEvent(wx.wxEVT_BUTTON))
+
+
+class BasicOverlayButton(Button):
+    """ Standard overlay button, partially transparent with more opaque hover mode """
+
+    def __init__(self, path_or_image, position):
+        self._opaque = False
+
+        if isinstance(path_or_image, Image.Image):
+            image = path_or_image
+        else:
+            image = Image.open(path_or_image)
+
+        super().__init__(position, image.size, image, image)
+
+        self.original_image = image
+        self.default_image = image
+
+    @Button.default_image.setter
```

Comment:
To be clear, this is wrapping `default_image.setter` with the `super`'s `default_image.setter` method, yes? That's neat!

Reply:
It's essentially overriding the `default_image` setter from the parent class.

Reply:
So why use `@Button.default_image.setter` over `@default_image.setter`? Is there an effective difference?

Reply:
Yes: because the latter didn't work 😈

Reply:
Or rather because I didn't re-implement the getter. If it'd done that, then it probably would work. This was just a way to do it and _not_ reimplement the getter.

Reply:
I suppose that makes sense, since `BasicOverlayButton` doesn't have that property, it doesn't know about the `Button.default_image.setter`. I'm good with it.